### PR TITLE
Change remaining modules to assertk

### PR DESCRIPTION
**Pull request by wmontwe**
This changes all remaining modules to use assertk assertions and remove truth library from the project.

Depends on: #6769

Fixes: #6755

### CODE REVIEW

File: app/core/src/test/java/com/fsck/k9/notification/NewMailNotificationManagerTest.kt [hunk 1]
```diff
@@ -173,7 +183,8 @@ class NewMailNotificationManagerTest {
         }
 
         assertNotNull(result) { data ->
-            assertThat(data.cancelNotificationIds).containsExactly(
+            assertThat(data.cancelNotificationIds).containsExactlyInAnyOrder(
+                // FIXME is the order important? If yes, use containsExactly
```
Comment (wmontwe) [inline app/core/src/test/java/com/fsck/k9/notification/NewMailNotificationManagerTest.kt:187]: I had to change this test to `containsExactlyInAnyOrder` as the order doesn't match the expected one. Is the order important?

File: app/storage/src/test/java/com/fsck/k9/storage/messages/DeleteMessageOperationsTest.kt [hunk 2]
```diff
@@ -184,6 +190,6 @@ class DeleteMessageOperationsTest : RobolectricTest() {
 
         val threads = sqliteDatabase.readThreads()
         assertThat(threads).hasSize(2)
-        assertThat(threads.map { it.id }.toSet()).isEqualTo(setOf(threadId1, threadId2))
+        assertThat(threads.map { it.id }).containsExactly(threadId1, threadId2)
```
Comment (cketti) [inline app/storage/src/test/java/com/fsck/k9/storage/messages/DeleteMessageOperationsTest.kt:193]: This should be `containsExactlyInAnyOrder`

File: backend/imap/src/test/java/com/fsck/k9/backend/imap/ImapSyncTest.kt [hunk 3]
```diff
@@ -151,7 +157,8 @@ class ImapSyncTest {
 
         imapSync.sync(FOLDER_SERVER_ID, defaultSyncConfig, syncListener)
 
-        assertThat(backendFolder.getMessageServerIds()).containsExactly("9", "13")
+        // FIXME Does the order matter? If yes, we need to fix ImapFolder: containsExactly -> containsOnly
```
Comment (cketti) [inline backend/imap/src/test/java/com/fsck/k9/backend/imap/ImapSyncTest.kt:160]: Truth's `containsExactly` doesn't check the item order. So it's generally a mistake to replace it with assertk's `containsExactly`. Most of the time it should probably be replaced by `containsExactlyInAnyOrder`.

Also, if a method returns a `Set` the order generally doesn't matter. If it ever does, we'll probably return a special subclass of `Set`.

File: backend/imap/src/test/java/com/fsck/k9/backend/imap/ImapSyncTest.kt
Comment (cketti) [inline backend/imap/src/test/java/com/fsck/k9/backend/imap/ImapSyncTest.kt:1]: The added `toList()` calls in this file should all be unnecessary.

File: app/storage/src/test/java/com/fsck/k9/storage/messages/DeleteFolderOperationsTest.kt [hunk 4]
```diff
@@ -43,8 +47,10 @@ class DeleteFolderOperationsTest : RobolectricTest() {
         assertThat(messages.first().uid).isEqualTo("message2")
 
         val messagePartFiles = messagePartDirectory.listFiles()
-        assertThat(messagePartFiles).hasLength(1)
-        assertThat(messagePartFiles!!.first().name).isEqualTo(messagePartId.toString())
+        assertThat(messagePartFiles).isNotNull().all {
+            hasSize(1)
+            transform { it.first().name }.isEqualTo(messagePartId.toString())
+        }
```
Comment (cketti) [inline app/storage/src/test/java/com/fsck/k9/storage/messages/DeleteFolderOperationsTest.kt:53]:
```suggestion
        assertThat(messagePartFiles).isNotNull()
            .extracting { it.name }.containsExactly(messagePartId.toString())

```

File: app/core/src/test/java/com/fsck/k9/message/html/UriMatcherTest.kt [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        var i = 0
+        val end = uris.size
+        while (i < end) {
+            val uri = uris[i]
+            val startIndex = text.indexOf(uri)
+            assertThat(startIndex).isNotEqualTo(-1)
+            val (startIndex1, endIndex, uri1) = uriMatches[i]
+            assertThat(startIndex1).isEqualTo(startIndex)
+            assertThat(endIndex).isEqualTo(startIndex + uri.length)
+            assertThat(uri1).isEqualTo(uri)
```
Comment (cketti) [inline app/core/src/test/java/com/fsck/k9/message/html/UriMatcherTest.kt:102]:
```suggestion
            val uriMatch = uriMatches[i]
            assertThat(uriMatch.startIndex).isEqualTo(startIndex)
            assertThat(uriMatch.endIndex).isEqualTo(startIndex + uri.length)
            assertThat(uriMatch.uri).isEqualTo(uri)
```

File: app/storage/src/test/java/com/fsck/k9/storage/messages/DeleteMessageOperationsTest.kt [hunk 6]
```diff
@@ -184,6 +190,6 @@ class DeleteMessageOperationsTest : RobolectricTest() {
 
         val threads = sqliteDatabase.readThreads()
         assertThat(threads).hasSize(2)
-        assertThat(threads.map { it.id }.toSet()).isEqualTo(setOf(threadId1, threadId2))
+        assertThat(threads.map { it.id }).containsExactlyInAnyOrder(threadId1, threadId2)
```
Comment (cketti) [inline app/storage/src/test/java/com/fsck/k9/storage/messages/DeleteMessageOperationsTest.kt:193]: Could be replaced by:

```kotlin
assertThat(threads).extracting { it.id }.containsExactlyInAnyOrder(threadId1, threadId2)
```

File: app/core/src/test/java/com/fsck/k9/message/html/UriMatcherTest.kt (hunk 5 above)
Comment (wmontwe) [reply app/core/src/test/java/com/fsck/k9/message/html/UriMatcherTest.kt:102]: Thats the Java to Kotlin conversation that likes to use destructuring.

**Pull request merged**